src/mame/apple/apple3_v.cpp

```diff
@@ -219,44 +209,50 @@ void apple3_state::text80(bitmap_ind16 &bitmap, const rectangle &cliprect)
 			uint8_t const *char_data;
 			pen_t fg, bg;
 
+			fg = GREEN;
+			bg = BLACK;
+
 			/* first character */
 			ch = ram[offset + 0x0000];
 			char_data = &m_char_mem[(ch & 0x7F) * 8];
-			fg = (ch & 0x80) ? GREEN : BLACK;
-			bg = (ch & 0x80) ? BLACK : GREEN;
 
 			for (int row = 0; row < 8; row++)
 			{
 				for (int col = 0; col < 7; col++)
 				{
-					int lc;
-					if (m_smoothscr)
+					uint16_t *const dest = &bitmap.pix(y * 8 + row, x * 14 + col + 0);
+
+					// Flash or Inverse
+					if ((!(ch & 0x80) && (char_data[row] & 0x80) && m_flash) || (!(ch & 0x80) && !(char_data[row] & 0x80)))
 					{
```

Comment:
Same here – it can use the same minimised expression.

Reply:
updated